src/Transport/Receiving/MessageExtensions.cs

```diff
@@ -40,5 +42,24 @@ public static string GetMessageId(this ServiceBusReceivedMessage message)
 
             return message.MessageId;
         }
+
+        public static BinaryData GetBody(this ServiceBusReceivedMessage message)
+        {
+            var body = message.Body ?? new BinaryData(Array.Empty<byte>());
+            var memory = body.ToMemory();
+
+            if (!memory.IsEmpty && message.ApplicationProperties.TryGetValue(TransportMessageHeaders.TransportEncoding, out var value) && value.Equals("wcf/byte-array"))
```

Comment:
I think it would have been beneficial to have a comment here about why we are doing this since this code can be removed May 2022 when the transport is out of support right?

Reply:
I don't think the code can ever be removed. While will no longer provide bug fixes/releases for legacy ASB, we will still be committed to wire compatibility with endpoints running older versions of the software.